apps/server/src/infra/etherpad-client/etherpad-client.adapter.ts

```diff
@@ -79,86 +80,77 @@ export class EtherpadClientAdapter {
 		groupId: GroupId,
 		authorId: AuthorId,
 		parentId: EntityId,
-		sessionCookieExpire: Date
+		sessionCookieExpire: Date,
+		durationThreshold: number
 	): Promise<SessionId> {
-		let sessionId: SessionId | undefined;
-		sessionId = await this.getSessionIdByGroupAndAuthor(groupId, authorId);
+		const session = await this.getSessionByGroupAndAuthor(groupId, authorId);
 
-		if (sessionId) {
-			return sessionId;
+		if (session && this.isSessionDurationSufficient(session, durationThreshold)) {
+			return session.id;
 		}
 
 		const response = await this.tryCreateSession(groupId, authorId, sessionCookieExpire);
-		const session = this.handleEtherpadResponse<InlineResponse2004>(response, { parentId });
+		const newSession = this.handleEtherpadResponse<InlineResponse2004>(response, { parentId });
 
-		sessionId = EtherpadResponseMapper.mapToSessionResponse(session);
+		const sessionId = EtherpadResponseMapper.mapToSessionResponse(newSession);
 
 		return sessionId;
 	}
 
+	private isSessionDurationSufficient(session: Session, durationThreshold: number): boolean {
+		const nowUnixTimestamp = Math.floor(new Date(Date.now()).getTime() / 1000);
```

Comment:
nowUnixTimestamp in which unit? nowUnixTimestampSeconds ? 
A timestemps is always in ms.